packages/apple-pay-integration/src/apple-pay-customer-strategy.ts

```diff
@@ -76,11 +76,14 @@ export default class ApplePayCustomerStrategy implements CustomerStrategy {
         this._onAuthorizeCallback = onPaymentAuthorize;
         this._onError = onError;
 
-        await this._paymentIntegrationService.loadPaymentMethod(methodId);
-
-        const state = this._paymentIntegrationService.getState();
+        let state = this._paymentIntegrationService.getState();
 
-        this._paymentMethod = state.getPaymentMethodOrThrow(methodId);
+        try {
+            this._paymentMethod = state.getPaymentMethodOrThrow(methodId);
+        } catch (_e) {
+            state = await this._paymentIntegrationService.loadPaymentMethod(methodId);
+            this._paymentMethod = state.getPaymentMethodOrThrow(methodId);
```

Comment:
What if it fails again?

Reply:
Then should be similar experience as of now, as this statement wasn't wrapped in try catch before as well.